http/server.lua

```diff
@@ -15,6 +16,21 @@ local errno = require 'errno'
 
 local DETACHED = 101
 
+-- A table that maps fiber object to time when it was active last time.
+-- Using for tracking idle connections.
+local connections = {}
+
+local function close_idle_connections(connections, idle_timeout)
+    -- FIXME: complexity is O(N) and it is bad
+    for fiber_obj, start_time in pairs(connections) do
+        if fiber.clock() - start_time >= idle_timeout then
+            log.info("Connection closed, idle_timeout was exceeded (fiber id %d)\n", fiber_obj:id())
```

Comment:
We have log_errors and log_requests callbacks. So I guess it's a bit inconsistent to use pure built-in log.

Reply:
Done.

```lua
diff --git a/http/server.lua b/http/server.lua
index 6b58f9d..5b80de5 100644
--- a/http/server.lua
+++ b/http/server.lua
@@ -20,7 +20,7 @@ local function close_idle_connections(connections, idle_timeout)
     -- FIXME: complexity is O(N) and it is bad
     for fiber_obj, start_time in pairs(connections) do
         if fiber.clock() - start_time >= idle_timeout then
-            log.info("Connection closed, idle_timeout was exceeded (fiber id %d)\n", fiber_obj:id())
+            logreq("Connection closed, idle_timeout was exceeded (fiber id %d)\n", fiber_obj:id())
             fiber_obj:cancel()
             connections[fiber_obj] = nil
         end
@@ -768,10 +768,12 @@ end
 local function process_client(self, s, peer)
     while true do
         local hdrs = ''
+        local route = self:match(p.method, p.path)
+        local logreq = get_request_logger(self.options, route)
 
         if self.options.idle_timeout > 0 then
             connections[fiber.id] = fiber.clock()
-            close_idle_connections(self.connections, self.options.idle_timeout)
+            close_idle_connections(self.connections, self.options.idle_timeout, logreq)
         end
 
         local is_eof = false
@@ -815,8 +817,6 @@ local function process_client(self, s, peer)
             s:write('HTTP/1.0 100 Continue\r\n\r\n')
         end
 
-        local route = self:match(p.method, p.path)
-        local logreq = get_request_logger(self.options, route)
         logreq("%s %s%s", p.method, p.path,
                p.query ~= "" and "?"..p.query or "")
 ```